### PR TITLE
Update github release workflow

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
$subject.

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -14,16 +12,44 @@ jobs:
...
         run: |
-          NEW_VERSION=$(echo "${GITHUB_REF}" | cut -d "/" -f3)
-          echo "New version: ${NEW_VERSION}"
-          echo "Github username: ${GITHUB_ACTOR}"
-          ./gradlew -PpublishToCentral=true -Pversion=${NEW_VERSION} publish
+          ./gradlew release -Prelease.useAutomaticVersion=true
+          ./gradlew -Pversion=${VERSION} publish -x test
+      - name: Create Github release from the release tag
+        run: |
+          curl --request POST 'https://api.github.com/repos/ballerina-platform/module-ballerina-sql/releases' \
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:42]:
```suggestion
          curl --request POST 'https://api.github.com/repos/ballerina-platform/module-ballerina-xslt/releases' \
```

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -14,16 +12,44 @@ jobs:
...
+          --header 'Accept: application/vnd.github.v3+json' \
+          --header 'Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}' \
+          --header 'Content-Type: application/json' \
+          --data-raw '{
+             "tag_name": "v'"$VERSION"'",
+             "name": "module-ballerina-sql-v'"$VERSION"'"
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:48]:
```suggestion
             "name": "module-ballerina-xslt-v'"$VERSION"'"
```

**Review by niveathika — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -14,16 +12,44 @@ jobs:
...
+          --data-raw '{
+             "tag_name": "v'"$VERSION"'",
+             "name": "module-ballerina-xslt-v'"$VERSION"'"
+          }'
+        - name: Post release PR
+          env:
+            GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+          run: |
+            curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
+            bin/hub pull-request -m "[Automated] Sync master after "$VERSION" release"
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:55]: wrong indentation

Comment (Bhashinee) [reply .github/workflows/publish-release.yml]: Fixed indentation

**Review by niveathika — APPROVED** [review]

**Pull request merged**